pkg/wasp/pod-ranker/rank_utils.go

```diff
@@ -134,7 +135,23 @@ func exceedMemoryLimits(summary summaryFunc) cmpFunc {
 func hasContainerExceedMemoryLimits(pod *v1.Pod, summary stats_collector.PodSummary) bool {
 	for _, container := range pod.Spec.Containers {
 		if rQuantity, ok := container.Resources.Limits[v1.ResourceMemory]; ok {
-			memoryAndSwapSumQuantity := memoryAndSwapUsage(*summary.Containers[container.Name].MemoryWorkingSetBytes, *summary.Containers[container.Name].MemorySwapCurrentBytes)
+			containerStats, exists := summary.Containers[container.Name]
```

Comment:
@Barakmor1 I could avoid code duplication here by appending the regular containers and init containers into a single list, but I prefer to do it in a follow-up, especially that we lack tests to protect us from re-factoring